tokio-postgres/src/connect.rs

```diff
@@ -32,18 +41,28 @@ where
             .copied()
             .unwrap_or(5432);
 
+        // The value of host is always used as the hostname for TLS validation.
         let hostname = match host {
             Host::Tcp(host) => host.as_str(),
             // postgres doesn't support TLS over unix sockets, so the choice here doesn't matter
             #[cfg(unix)]
             Host::Unix(_) => "",
         };
-
         let tls = tls
             .make_tls_connect(hostname)
             .map_err(|e| Error::tls(e.into()))?;
 
-        match connect_once(host, port, tls, config).await {
+        // If both host and hostaddr are specified, the value of hostaddr is used to to establish the TCP connection.
+        let hostaddr = match host {
+            Host::Tcp(_hostname) => match config.hostaddr.get(i) {
+                Some(hostaddr) if hostaddr.is_empty() => Host::Tcp(hostaddr.clone()),
```

Comment:
Isn't this checking if the hostaddr string is present but empty?

Reply:
Urgh. Thanks. I made this probably  due to resolving a clippy recommendation from != to !is_empty.